fix: show correct badge count on notebook button

daibhin opened the pull request:
## Problem

Reported in https://posthog.slack.com/archives/C05N9R3HT7V/p1695922124646429. The badge was showing the count for all the notebooks containing an insight, and not making use of the saved insights id

## Changes

- Pass the saved insights id to the `NotebookSelectButton`
- Use a special `id_match_structure` to find matching insights

## How did you test this code?

Updated tests

Commit:
fix: show correct badge count on notebook button

Commit:
update tests

pauldambra reviewed (approved):
🙌

Commit:
remove final insight reference

benjackwhite commented on frontend/src/scenes/notebooks/NotebookSelectButton/NotebookSelectButton.tsx:
```diff
@@ -104,7 +104,7 @@ export function NotebookSelectList(props: NotebookSelectProps): JSX.Element {
 
     return (
         <div className="space-y-2 flex flex-col flex-1 h-full overflow-hidden">
-            <div className="border-b space-y-2 flex-0">
+            <div className="border-b space-y-2 flex-0 pb-2">
```
We have space-y-2 and pb-2 conflicting with each other. Fine if its necessary just something that normally makes me think, one of the values needs deleting and the other changing

benjackwhite commented on posthog/api/notebook.py:
```diff
@@ -281,6 +281,8 @@ def _filter_request(self, request: request.Request, queryset: QuerySet) -> Query
                             # replay timestamps are not at the top level, they're one-level down in a content array
                             presence_match_structure = [{"content": [{"type": f"ph-{target}"}]}]
                             id_match_structure = [{"content": [{"attrs": {"sessionRecordingId": match}}]}]
+                        elif target == "query":
```
🤔 So we are making `query` only work for a specific Insight? This feels very off...

If this is the way to go then I think the "target" should explicitly be `saved-insight` or something.

daibhin replied:
My understanding was that this is just for cases in which there is an id present e.g. when the search query looks like `query:vk7xz3` where the content after the colon is the short id of the saved insight. You could still search notebooks that contain / do not contain queries with the search param `query:true` or `query:false`.

The only reason we special case this lookup is because the `NotebookNodeQuery` does not have an `id` attribute which the search endpoint assumes. Its id is nested inside the `query` attribute.

Does that explain your question?

daibhin replied:
@pauldambra am I right in thinking that this will still allow you to search for the presence of a query with the `query:true` param?

pauldambra replied:
so long as this test is still valid

```
        # filter by insight
        insight_filter_response = self.client.get(f"/api/projects/{self.team.id}/notebooks?contains=query:true")
        assert [n["id"] for n in insight_filter_response.json()["results"]] == [insight_content_notebook]
```

then yep we can filter by presence of "type" and presence of an instance of a "type"

Commit:
Merge branch 'master' into dn-fix/insight-notebook-badge

Commit:
cleanup button styling

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

robbie-c committed:
feat(web-analytics): Fix top pages sql and add some extra insights (#17850)

* Fix top pages SQL

* Use person id for now

* Add limits to web analytics queries

* Add some trends to web analytics

* Tweak query date ranges and formatting

* Consistently use person_id for now

daibhin committed:
fix: remove content width toggle from popover (#17775)

tomasfarias committed:
fix(batch-exports): Include properties in feature_enabled call (#17860)

bretthoerner committed:
chore: set Sentry release using commit.txt (#17855)

tiina303 committed:
chore: Remove promise manager usage from ingestion warnings & db (#17650)

ahsan722505 committed:
fixes "CSV export is limited to 10000 events" menu is hidden behind export menu (#17834)

raquelmsmith committed:
feat: onboarding, remove product intro, fix navigation (#17858)

* move onstart to products page

* remove product intro

* include stepkey as prop

* redo how navigation works

* go back to products if first step

* fix url redirect loop

* fix jitteryness with otherproductsstep

* fix grumpy linting

daibhin committed:
feat: sidebar widgets (#17859)

bretthoerner committed:
chore(plugin-server): kafka: change analytics linger.ms and batch.siz… (#17820)

* chore(plugin-server): kafka: change analytics linger.ms and batch.size for all producers

* add envvars

liyiy committed:
feat(surveys): optional questions for multipleq (#17830)

posthog-bot committed:
chore(deps): Update posthog-js to 1.83.0 (#17871)

neilkakkar committed:
feat(surveys): Add question html support (#17847)

robbie-c committed:
test: Increase playwright timeouts (#17872)

* Increase playwright timeouts

* Tidy up comments

thmsobrmlr committed:
fix(celery): celery compatibility with importlib (#17874)

thmsobrmlr committed:
fix(insights): do not remove empty series from converted queries (#17879)

benjackwhite committed:
feat: Person node changes (#17880)

Person node changes

tiina303 committed:
chore: Remove runner abstraction from onEvent (#17683)

MarconLP committed:
fix: update slack description (#17884)

benjackwhite committed:
feat: Reworked Playlist UI with Notebook support (#17802)

raquelmsmith committed:
feat: clarify experiments usage in FFs onboarding (#17870)

clarify experiments usage in FFs onboarding

bretthoerner committed:
chore(plugin-server): run node-rdkafka with cooperative rebalancing patched in (#17747)

* Remove node-rdkafka-acosom

* Add node-rdkafka

* Replace node-rdkafka-acosom imports with node-rdkafka

* Patch node-rdkafka with changes from https://github.com/PostHog/node-rdkafka/

* Add patch directions to README

mariusandra committed:
fix(insights): loading in saved insights card view (#17875)

Twixes committed:
chore(invites): Clarify expiration message (#17862)

* chore(invites): Clarify expiration message

* Change wording

daibhin committed:
fix: disable link button until full href (#17889)

daibhin committed:
fix: remove gap cursor (#17888)

bretthoerner committed:
chore(plugin-server): trigger ingestion deploy (#17900)

frankh committed:
feat: Set keepalive to 60 on gunicorn (#17897)

Set keepalive to 60 on gunicorn

The default is 2 seconds, the default for ALBs is 30 seconds

This can cause a race condition where gunicorn closes the connection
as the ALB sends a request, resulting in a 502.

benjackwhite committed:
feat: Easier selecting of next line after node (#17883)

pauldambra committed:
chore: launch celery by module in pycharm (#17901)

* chore: launch celery by module in pycharm

* Revert "chore: launch celery by module in pycharm"

This reverts commit 6a6d3b5274cbf65e24d0f9c68910c77c520aa59f.

* run in vscode more like in production

benjackwhite committed:
perf: Reworked types and rendering to try and reduce needless renders (#17895)

daibhin committed:
chore: remove title expand toggle (#17887)

jurajmajerik committed:
feat(surveys): add Rating question results viz (#17886)

benjackwhite committed:
feat(3000): Added link specific styles (#17867)

benjackwhite committed:
fix: Notebook colors and start position (#17907)

benjackwhite committed:
fix: Insight tooltip overscrolling issue (#17906)

Fix tooltip overscrolling issue

thmsobrmlr committed:
fix(dev): do not sync the cloud announcement flag (#17911)

neilkakkar committed:
chore(decide): Add missing span instrumentation (#17910)

tiina303 committed:
chore: skip flaky test in session-recordings-consumer-v2.test.ts (#17896)

benjackwhite committed:
chore: Update install script (#17851)

Fixed script

thmsobrmlr committed:
fix(types): fix value inspector button warning (#17876)

Co-authored-by: Michael Matloka <dev@twixes.com>

Twixes committed:
chore(dev): Allow generating demo data for existing projects (#17913)

thmsobrmlr committed:
fix(autocapture): add taxonomic filter key as data-attr (#17899)

daibhin committed:
fix: account for negative timestamps from rrweb (#17849)

pauldambra committed:
chore: yeet CH recordings ingestion (#17572)

Removing ClickHouse based recordings

One big yeet for a man, a great yeet for humanity

frankh committed:
feat: Add env var to skip migration checks (#17915)

Add env var to skip migration checks

This will allow us to skip the checks in kubernetes pods that use
initContainers but without effecting existing deploys or docker-compose

mariusandra committed:
feat(hogql): events query runner (#17892)

bretthoerner committed:
chore(plugin-server): use existing INGESTION_BATCH_SIZE and KAFKA_CON… (#17927)

chore(plugin-server): use existing INGESTION_BATCH_SIZE and KAFKA_CONSUMPTION_SESSION_TIMEOUT_MS

neilkakkar committed:
chore: Remove stale feature flags (#17925)

benjackwhite committed:
fix: Editor initial content (#17912)

daibhin committed:
feat: persons & events settings in notebooks (#17866)

timgl committed:
chore: Remove org-wide retro from sprint planning (#17928)

pauldambra committed:
fix: throw error so test failures are not swallowed (#17926)

So obvious in retrospect

If you listen to the Cypress fail event without re-throwing then you swallow all test failures

"fun"

When reviewing #17919 I knew the Cypress tests would have to be failing which is what prompted me to check

Introduced in bbb7ed9 (July 10th!)

benjackwhite committed:
fix: Additional parsing of incoming messages (#17937)

benjackwhite committed:
chore: Remove v2 suffix for replay consumer (#17938)

robbie-c committed:
feat(css): Add tailwind size breakpoints (#17918)

* feat(css): Add tailwind size breakpoints

* Only do breakpoint expansion for grid css classes

benjackwhite committed:
feat: Respect resize of DataTable in Notebook (#17929)

mariusandra committed:
feat(hogql): query debugger (#17939)

xvello committed:
chore(plugins-server): remove eachBatchLegacyIngestion (#17894)

benjackwhite committed:
fix: Check for the event generally being truthy (#17942)

benjackwhite committed:
feat: Added notebooks to sidebar (#17919)

mariusandra committed:
fix(hogql): quick fix for events query index (#17941)

pauldambra committed:
chore: twiddle with logging (#17944)

* chore: twiddle with logging

* more specifically:

pauldambra committed:
fix: e2e testing posthog capture (#17940)

* fix: e2e testing posthog capture

* wat

* wat

* wat:

jurajmajerik committed:
feat(surveys): Single Choice question results (#17923)

pauldambra committed:
chore: update wait-on-action (#17945)

bretthoerner committed:
chore(plugin-server): add labels to merge txn metrics (#17934)

robbie-c committed:
feat: Add django command for geoip (#17947)

* Add django command for geoip

* Support multiple ips in one command

* Add pretty printing and the cache enum

Commit:
fix: early access management cancel button (#17857)

benjackwhite committed:
feat: Allow pausing whilst buffering (#17951)

benjackwhite committed:
fix: Performance issue with player logs (#17948)

neilkakkar committed:
fix(flags): Fix spinner on new feature flag (#17953)

Gilbert09 committed:
feat: Added lazy joins for groups on events (#17950)

* Added lazy joins for groups on events

* Update query snapshots

* Fixed test_resolver.py tests

* Added aliases for groups from group mappings

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

benjackwhite committed:
feat: Notebook feedback changes (#17952)

robbie-c committed:
feat(web-analytics): Add web analytics filters - Part 2 (#17878)

* Add PropertiesFilters to web dashboard

* Add some property filters for web analytics queries

* Add titles

* Get working pathname filtering

* Add clickable pathname filtering

* Slight pretty some columns

* Finish making it look nice on mobile

* Update query snapshots

* Update query snapshots

* Fix loading bars going on top of UI elements

* Fix typing problems

* Remove the unused isNull functions

* Update posthog/hogql/property.py

Co-authored-by: Marius Andra <marius.andra@gmail.com>

* Add extra check for filter type

* Add missing import

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: Marius Andra <marius.andra@gmail.com>

pauldambra committed:
chore: deflake a test (#17956)

benjackwhite committed:
feat: Added not found helper for notebooks (#17957)

daibhin committed:
fix: margin finness & insertion suggestion height (#17955)

thmsobrmlr committed:
fix(design): remove background from action filter (#17964)

mariusandra committed:
feat(hogql): query modifiers (#17946)

Gilbert09 committed:
fix: ensure we dont overwrite events fields with group mappings (#17965)

Ensure we dont overwrite events fields with group mappings

robbie-c committed:
feat(web-analytics): Fix source query views and uniques - remove blended source (#17961)

Fix source query views and uniques

benjackwhite committed:
feat: Add basic Group / cohort Notebok nodes (#17962)

neilkakkar committed:
feat(surveys): Add cypress e2e tests (#17920)

benjackwhite committed:
feat: Node title placeholders with custom title (#17966)

mariusandra committed:
feat(database): faster queries from the persons table (#17811)

raquelmsmith committed:
chore: send reports closer to midnight (#17960)

send at 12:05 am

Twixes committed:
perf: Parse HogQL with C++ for a huge speedup (#17659)

* Add partial C++ parser of HogQL

* Support all the rules and add proper error handling

* Use `AlignAfterOpenBracket: BlockIndent`

* Reuse all the parser tests for the C++ backend

* Fix relationship between headers and implementations

* Add more error handling and fix minor issues

* Build both Python and C++ parsers in the package.json script

* Update ARRAY JOIN error assertion

* Improve timeit

* Move the C extension to the top level

* Refactor `vector_to_list_string`

* Build the parser on Linux

* Build wheels for the parser

* Simplify Linux build and fix macOS

* Fix Homebrew paths on x86 and don't fail fast in CI

* Set MACOSX_DEPLOYMENT_TARGET for C++20

* Set up QEMU for Linux ARM builds

* Publish the wheels on PyPI

* Avoiding Linux ARM emulation in CI for now

* Build sdist too

* Revert Dockerfile changes

* Fix PyPI publish

* Add README and optimize sdist build

* Use setup.py directly instead of build

* Use PyPI hogql-parser instead of local

* Also revert production-unit.Dockerfile

* Fix sdist upload and add Linux ARM back

* No Linux ARM build in the end

* Fix artifact uploading

* Do try building Linux ARM

We need this for prod.

* Use `npm` in `grammar:build`

`pnpm` is not available in that job.

* Fix formatting of hogql_parser

* Build everything on macOS

* Revert "Build everything on macOS"

Not so fast actually.

* Use hogql-parser=0.1.1

* Fix dylib in macOS wheel

* Bump hogql-parser version

* Fix missing module error

* Delete timeit.py

* Make error handling robust

* Format the C++

* Use `hogql-parser==0.1.1`

* Fix reserved keyword error assertions

* Use HEAD hogql_paresr in CI

* Fix `apt` usage

* Add some sudo in CI

* Ensure package will be releasable before build

* Bump version to 0.1.3

* Cover C++ `unquote_string` with tests

* Use BuildJet ARM runners for ARM builds

* Add some instructions

* Add HogQL version check to backend CI

* Update requirements.txt

* Use `setuptools` instead of the deprecated `distutils`

* Fix working dir in backend CI

* Align ANTLR versions

* Add test for "mismatched input"

This is thrown differently than other HogQLSyntaxExceptions in C++, so might help reveal what's going on with tests failing only on Linux CI and not macOS dev

* Add types and bump version

* Comment instead of failing version check

* Automate hogql-release version bump

* Fix checkout token

* Don't build hogql-parser if there were no changes

* Update query snapshots

* Update query snapshots

* Update query snapshots

* Update query snapshots

* Improve documentation

* Use new hogql-parser version

* Fix error start and end initialization

* Note `antlr4-cpp-runtime`

Co-authored-by: Marius Andra <marius.andra@gmail.com>

* Also remove NUL chars in C++

* Check ANTLR4 runtime archive checksum for security

* Note more decrefs to add

* Add vector size checks

* Use new hogql-parser version

* Don't support the `start` arg in C++ `parse_expr`

* Use new hogql-parser version

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: Marius Andra <marius.andra@gmail.com>

Gilbert09 committed:
feat: Added the first iteration of trend insghts breakdowns (#17891)

* Added the first iteration of trend insghts breakdowns

* Removed team id from all events filters

* Added support for histogram breakdowns

* PR fixes

* Support group breakdowns

* Abstract property chain into a cached prop

mariusandra committed:
feat(hogql): debugger split view (#17970)

Gilbert09 committed:
chore: add timings measure for breakdown values query (#17974)

Add timings measure for breakdown values query

daibhin committed:
chore: update copy to reflect popover over pinning (#17971)

mariusandra committed:
feat(hogql): intelligent argmax on persons table (#17975)

bretthoerner committed:
chore(plugin-server): move runner process.env calculations to Hub (#17933)

tiina303 committed:
chore: Update capabilities.test.ts (#17465)

jurajmajerik committed:
feat(surveys): Multiple Choice question results (#17954)

mariusandra committed:
feat(hogql): edit SQL directly where possible (#17905)

Twixes committed:
style(3000): Use new Central icons (#16463)

* style(3000): Use new Central icons

* Use prefixed icons

* Use new icons

* Remove `width` rule

* Change Lemon `IconNotebook` to Material

* Fix icon width

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (2)

* Fix notebook icon coloring

* Update UI snapshots for `chromium` (2)

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

pauldambra committed:
chore: update snapshots for optimise aggregation setting (#17990)

tomasfarias committed:
feat(temporal-worker): Add steps to trigger deployment (#17881)

tomasfarias committed:
chore(batch-exports): Minor model docstring updates (#17993)

thmsobrmlr committed:
refactor(insights): remove query from editor filter props (#17963)

pauldambra committed:
fix: don't process subscriptions with no assets (#17988)

thmsobrmlr committed:
refactor(insight): simplify insightVizDataLogic update actions (#17972)

pauldambra committed:
fix: console log fiddling (#17996)

* fix: use kea for debouncing

* explicitly capture exception

smallbrownbike committed:
feat(surveys): Improved survey creation flow (#17902)

* create edit survey components for feature flagging

* visual type picker

* organization

* move previews into tooltips

* targeting

* sync preview and accordion

* better presentation view

* Update UI snapshots for `chromium` (2)

* fix migration file lint

* Update UI snapshots for `chromium` (2)

* make collapse panel header easier to see

* label question index

* more fixes

* update

* surveys new creation flow flag

* fix linter spacing

* form appearance file

* Update UI snapshots for `chromium` (2)

* fix

* fix

* Update UI snapshots for `chromium` (2)

* fix targeting

* fix cypress tests

* black

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: Li Yi Yu <li@posthog.com>
Co-authored-by: Neil Kakkar <neilkakkar@gmail.com>

raquelmsmith committed:
feat: add and handle never drop data field (#17978)

* add and handle never drop data field

* Update query snapshots

* Update query snapshots

* Update query snapshots

* Update query snapshots

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

ivanagas committed:
fix: Nextjs onboarding snippets (#17979)

* Fix nextjs onboarding snippets

bretthoerner committed:
chore(plugin-server): remove INGESTION_DELAY_WRITE_ACKS and workerMethods (#17932)

* chore: stop using piscina worker methods for runEventPipeline

* chore(plugin-server): remove INGESTION_DELAY_WRITE_ACKS

---------

Co-authored-by: Tiina Turban <tiina303@gmail.com>

mariusandra committed:
feat(hogql-debug): add explain (#17998)

tiina303 committed:
feat: Move plugin options out of drawer + uninstall safety (#17980)

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

bretthoerner committed:
chore(plugin-server): remove some unused piscina references (#18012)

pauldambra committed:
feat: dedupe console logs (#18009)

* feat: dedupe console logs

* level

posthog-bot committed:
chore(deps): Update posthog-js to 1.83.1 (#18014)

liyiy committed:
feat(surveys): templates (#17904)

* routes for survey templates

* basic on click setup logic

* Update UI snapshots for `chromium` (2)

* import type

* Apply suggestions from code review

Co-authored-by: Joe Martin <84011561+joethreepwood@users.noreply.github.com>

* address feedback from cory

* fix logic

* remove unused logic file

* Update frontend/src/scenes/surveys/constants.tsx

Co-authored-by: Joe Martin <84011561+joethreepwood@users.noreply.github.com>

* Update frontend/src/scenes/surveys/constants.tsx

Co-authored-by: Neil Kakkar <neilkakkar@gmail.com>

* address comments

* Update UI snapshots for `chromium` (2)

* more fixes

* Update UI snapshots for `webkit` (2)

* Update UI snapshots for `webkit` (2)

* Update UI snapshots for `chromium` (2)

* move description and title together

* fix e2e

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: Joe Martin <84011561+joethreepwood@users.noreply.github.com>
Co-authored-by: Neil Kakkar <neilkakkar@gmail.com>

pauldambra committed:
chore: really capture console log search errors to Sentry (#18015)

liyiy committed:
chore(surveys): report template clicked (#18017)

pauldambra committed:
fix: console log query (#18021)

* fix: console log query

* Update query snapshots

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

pauldambra committed:
feat(plugin-server): allow to configure the rdkafka producer queue size (#18007)

* fix: allow larger batches for console log ingestion production

* not that

* queue size not batch size

daibhin committed:
feat: open settings when query inserted (#17969)

daibhin committed:
fix: text formatting icons in inline menu & slash commands (#17958)

daibhin committed:
fix: popover edits out of sync (#18006)

benjackwhite committed:
feat: Create Notebook from dashboard (#17968)

tiina303 committed:
feat: Separate out apps management tab (#18013)

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

tiina303 committed:
chore: Update plugin-scaffold to latest (#17930)

robbie-c committed:
feat(web-analytics): Add web analytics tabs (#17981)

* Add web analytics tabs

* Add stats table query runner and use it for pathname

* Add stats query runner, replacing page and source

* Remove classes for now-unused queries

* Add remaining breakdowns

* Prefer unknown to any

* Remove unused ctes

* Fix typing issues

* Fix python typing

* Switch to hogql filters for set_once properties

* Use cpp parser

* Move web analytics to the right navbar section

* Fix typing issues

pausan committed:
Fix/overlap taxonomic filter popup with scroll (#17987)

Gilbert09 committed:
fix: fixed missing param when building Breakdown (#17991)

Fixed missing param when building Breakdown

Gilbert09 committed:
feat: HogQL trends breakdown by hogql expressions (#17992)

* Fixed missing param when building Breakdown

* Added support for breakdowns with hogql expressions

neilkakkar committed:
fix: Make rating start at 0 for surveys (#18011)

tiina303 committed:
chore: Make pluginConfigs be keyed over config id not plugin id (#17999)

Twixes committed:
perf(hogql): Fix C++ parser leaks (#18022)

posthog-bot committed:
chore(deps): Update posthog-js to 1.83.2 (#18033)

liyiy committed:
fix(surveys): clean up survey edit code (#18010)

* fix(surveys): clean up survey edit code

* Update UI snapshots for `chromium` (2)

* add surveys template story

* Update UI snapshots for `chromium` (2)

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: Neil Kakkar <neilkakkar@gmail.com>

jurajmajerik committed:
feat(surveys): Implement Open Text visualization (#18016)

liyiy committed:
fix(surveys): fix survey template (#18034)

* fix(surveys): fix survey template

* fix appearance

* Update UI snapshots for `chromium` (2)

* make emoji colors darker

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

Gilbert09 committed:
feat: added trends breakdown by cohort (#18028)

* Fixed missing param when building Breakdown

* Added support for breakdowns with hogql expressions

* Added trends breakdown by cohort

* Remove None valued keys from the filter obj

tiina303 committed:
revert "chore: Make pluginConfigs be keyed over config id not plugin id" (#18037)

Revert "chore: Make pluginConfigs be keyed over config id not plugin id (#17999)"

This reverts commit df282f65107e2bb868d42f157f4545400c0d21e1.

Commit:
feat: show only login options if account exists when accepting an invite (#17917)

* add logic to redirect existing user

* add redirection on UI and user import

* backend tests

* fix linting issues

* add invite email to validate method

---------

Co-authored-by: Raquel Smith <raquelmsmith@users.noreply.github.com>

raquelmsmith committed:
fix: tiers show incorrect volumes if denominator differs (#18041)

* use correct denom for previous tier and show correct number

* test the func

raquelmsmith committed:
fix: react env var name inconsistency (#18044)

fix var name

liyiy committed:
fix(surveys): see multiple question surveys in overview (#18048)

Twixes committed:
fix(deps): Update @posthog/icons to fix React peer dependency (#18043)

raquelmsmith committed:
feat: show CTA for annual plan if over $1k/mo (#18049)

* show CTA for annual plan if over $1k/mo

* let it be dismissed

* Update UI snapshots for `chromium` (1)

* up front

* send event whn clicked

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

robbie-c committed:
fix(hogql): Fix boolean is not set (#18040)

mariusandra committed:
fix(queries): export limit (#18046)

mariusandra committed:
feat(hogql): use join for "in cohort" operations instead of subquery (#17354)

pauldambra committed:
feat: move buttons higher (#18039)

* feat: move buttons higher

* Update UI snapshots for `chromium` (2)

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

tiina303 committed:
chore: Remove install button from apps view (#18038)

Gilbert09 committed:
feat: added session duration as a breakdown option (#18056)

* Added session duration as a breakdown option

* Update query snapshots

* Fixed tests

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

posthog-bot committed:
chore(deps): Update posthog-js to 1.83.3 (#18035)

Co-authored-by: posthog-bot <posthog-bot@users.noreply.github.com>

neilkakkar committed:
fix(surveys): Add html descriptions for thank you (#18036)

mariusandra committed:
feat(queries): persons query source (#18060)

robbie-c committed:
feat(web-analytics): Basic adding of date ranges to the web dashboard (#18061)

Basic adding of date ranges to the web dashboard

liyiy committed:
chore(surveys): add surveys opt in to onboarding ingestion (#18045)

jurajmajerik committed:
fix(surveys): bypass HogQL coercion + UI fixes (#18063)

mariusandra committed:
chore(storybook): run a few times ¯\_(ツ)_/¯ (#18066)

posthog-bot committed:
chore(deps): Update posthog-js to 1.84.0 (#18069)

neilkakkar committed:
fix(surveys): Some UI/UX issues (#18070)

* fix more issues

* fix

* Update UI snapshots for `chromium` (2)

* update

* address comment

* Update UI snapshots for `chromium` (2)

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

raquelmsmith committed:
feat: surveys onboarding (#18053)

* first pass

* update sdks

* add stepkey

* fix typo

* use correct number of sig figs

jurajmajerik committed:
fix(surveys): check result exists (#18082)

neilkakkar committed:
fix(decide): Don't create exceptions for survey targeting flags (#18062)

robbie-c committed:
chore(insights): Making schema conversion from schema.ts to schema.py include everything in schema.ts (#18075)

* Convert all of schema.ts to schema.py

* Add explicit name to schema root

* Fix test

Gilbert09 committed:
feat(hogql): Added unit tests for the new trends query runner (#17730)

* Added unit tests for the new trends query runner

* Updated test file location

* Remove pydantic

* Update query snapshots

* Update query snapshots

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

tomasfarias committed:
feat(batch-exports): Add backfill workflow (#17909)

mariusandra committed:
feat(hogql): basic pretty print (#18086)

bretthoerner committed:
fix(plugin-server): wrap a few ingestion/consumer calls in simple ret… (#18072)

fix(plugin-server): wrap a few ingestion/consumer calls in simple retries

neilkakkar committed:
fix(version-checker): Use latest semantic version (#18085)

* fix(version-checker): Use latest semantic version

* elaborate

* more version nums for funsies

bretthoerner committed:
fix(plugin-server): clarify var name in retryIfRetriable (#18093)

raquelmsmith committed:
fix: redirect to correct view for surveys (#18081)

redirect to correct view for surveys

tomasfarias committed:
fix(batch-exports-backfill): Instantiate tuple when resuming from heartbeat (#18092)

posthog-bot committed:
chore(deps): Update posthog-js to 1.84.1 (#18096)

bretthoerner committed:
fix(plugin-server): consumer loop throwing trigger a shutdown (#18073)

* fix(plugin-server): join analytics ingestion server at top level

* Update plugin-server/src/main/pluginsServer.ts

Co-authored-by: Xavier Vello <xavier.vello@gmail.com>

---------

Co-authored-by: Xavier Vello <xavier.vello@gmail.com>

tomasfarias committed:
chore(batch-exports): Do not retry on forbidden (#18091)

Gilbert09 committed:
feat: allow placeholders in place of ratio expressions in hogql (#17973)

* Allow placeholders in place of ratio expressions in hogql

* Added C++ changes for RatioExpr

* Bump `hogql_parser` version

* Upgrade ANTLR version to align with brew latest

* Update query snapshots

* Fix parser change message body

The quotes weren't right, so the step failed instead of posting a comment.

* Revert "Update query snapshots"

This reverts commit 1302d3b24f138e36590cf5ff45769323379a4f95.

* Update MD5 sum

* Bump version again

* Use new hogql-parser version

---------

Co-authored-by: Michael Matloka <michal@matloka.com>
Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: Michael Matloka <dev@twixes.com>

xvello committed:
chore(p-s): add logs around setupPlugins (#18099)

bretthoerner committed:
chore(plugin-server): log timing per rdkafka batch loop rather than b… (#18100)

chore(plugin-server): log timing per rdkafka batch loop rather than based on a schedule

tomasfarias committed:
fix(batch-exports): Only read person properties for S3 (#18095)

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

bretthoerner committed:
fix(plugin-server): log EPS every 10 seconds, only log individual bat… (#18105)

fix(plugin-server): log EPS every 10 seconds, only log individual batches when they are slow

jurajmajerik committed:
feat(surveys): add responses to the events table + fixes (#18088)

raquelmsmith committed:
feat(billing): add menu option for reducing bill info (#18050)

add menu option for reducing bill info

raquelmsmith committed:
fix: support minimal top bar for onboarding screens (#18077)

tomasfarias committed:
fix(backfill-batch-exports): Do not run partial ranges (#18114)

mariusandra committed:
feat(hogql): mixed PoE mode (#18064)

* feat(hogql): mixed poe mode

* schema

pauldambra committed:
fix: update person recordings default date range (#18115)

tomasfarias committed:
chore(batch-exports): Do not retry on RefreshError and NotFound (#18107)

Twixes committed:
chore(hogql): Use the C++ parser in local dev (#18112)

* chore(hogql): Use the C++ parser in local dev

* Update all the parsing functions, not just `parse_select`

thmsobrmlr committed:
chore(deps): upgrade storybook to 7.5 (#18113)

bretthoerner committed:
fix(plugin-server): jitter and retry reloadPlugins (#18108)

bretthoerner committed:
fix(plugin-server): filter NaN out of buildIntegerMatcher (#18106)

Gilbert09 committed:
feat: added support for daily and weekly active user aggregations (#18101)

* Added support for daily and weekly active user aggregations

* Removed unused join

* Fixed tests and applied Sample placeholders

* Fixed formula running on an empty string

* Updated the name of QueryModifier

* Comments and fixed name casing

robbie-c committed:
feat(insights): Add dashboard filters to hogql insights (#18076)

* WIP

* Kinda sorta working

* Add dashboard filters to cache key

* Make mypy happy

* Add support for properties

* Use get_query_runner

* Add tests for dashboard_query

* Add tests for merging of filters

* Add test for hash when dashboard filters are applied

* Silently accept having no valid query runner

* Remove index type from DashboardFilter

* Fix type of dashboard_query

* Fix test typing

robbie-c committed:
feat(web-analytics): Add localised number formatting and right-aligned cells (#18116)

* Add better table formatting

* Use QueryContent to right-align table cells and headings

* Just use toLocaleString

bretthoerner committed:
fix(plugin-server): prevent multiple concurrent reloadPlugins and coa… (#18120)

fix(plugin-server): prevent multiple concurrent reloadPlugins and coalesce them when possible

tomasfarias committed:
fix(backfill-batch-exports): Account for jitter (#18121)

liyiy committed:
feat(surveys): auto opt in and out surveys for users (#18080)

* opt in surveys when creating survey and opt out when no active surveys

* update survey banner warning

* add auto opt ins for surveys list launch and stop surveys

* remove from onboarding step

* Update UI snapshots for `chromium` (2)

* update opt in from backend instead

* add filter for api surveys and post delete to receiver

* Update query snapshots

* fix popup flicker

* ambr snapshots

* disabled popup banner rename

* exclude api surveys in query instead

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

liyiy committed:
feat(surveys): gate premium features (#18094)

* pay gates wip

* Update UI snapshots for `chromium` (2)

* pay gate survey edit

* html descriptions

* Update UI snapshots for `chromium` (2)

* rename ctas in pay gate mini and use getavailablefeature

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `webkit` (2)

* feature flag pay gate

* Update UI snapshots for `chromium` (2)

* swap surveys beta tag for new and bump higher

* Update UI snapshots for `chromium` (1)

* fix merge conflict

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

liyiy committed:
fix(survey): fix question type placement and add open feedback template (#18125)

* move survey question type back up

* add open feedback template

* fix description

posthog-bot committed:
chore(deps): Update posthog-js to 1.84.2 (#18130)

rishabnayak committed:
ci(hogql): Don't run parser build workflow on forks (#18018)

* only run hogql parser workflow if running from upstream repo

* Fix capitalization

---------

Co-authored-by: Michael Matloka <michal@matloka.com>

dependabot[bot] committed:
chore(deps-dev): bump postcss from 8.4.18 to 8.4.31 (#17767)

Bumps [postcss](https://github.com/postcss/postcss) from 8.4.18 to 8.4.31.
- [Release notes](https://github.com/postcss/postcss/releases)
- [Changelog](https://github.com/postcss/postcss/blob/main/CHANGELOG.md)
- [Commits](https://github.com/postcss/postcss/compare/8.4.18...8.4.31)

---
updated-dependencies:
- dependency-name: postcss
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Commit:
fix: Notebook select matching (#17916)

Commit:
fix(insights): Move GF, GP and YT to top level on the world map (#17949)

Move GF, GP and YT to top level on the world map

Commit:
feat(insights): edit as sql (#18055)

Commit:
Merge branch 'master' into dn-fix/insight-notebook-badge

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

benjackwhite reviewed (approved):
👍 still has the issue of only really working for insight queries but i can live with that for now
